src/cmd/flux-exec.c

```diff
@@ -322,6 +327,35 @@ char *split_opt (const char *s, char sep, const char **val)
     return cpy;
 }
 
+static bool check_for_imp_run (int argc, char *argv[], const char **ppath)
+{
+    /* If argv0 basename is flux-imp, then we'll likely have to use
+     *  flux-imp kill to signal the resulting subprocesses
+     */
+    if (streq (basename (argv[0]), "flux-imp")) {
+        *ppath = argv[0];
+        return true;
+    }
+    return false;
+}
+
+static const char *flux_imp_path (flux_t *h)
+{
+    const char *imp = NULL;
+    flux_future_t *f;
+
+    if (!(f = flux_rpc (h, "config.get", NULL, FLUX_NODEID_ANY, 0))
+        || flux_rpc_get_unpack (f,
+                                "{s?{s?s}}",
+                                "exec",
+                                "imp", &imp) < 0)
+        fprintf (stderr, "error fetching config object: %s",
+                 future_strerror (f, errno));
+    flux_aux_set (h, NULL, f, (flux_free_f) flux_future_destroy);
+    return imp;
+}
```

Comment:
- `flux_` prefix in a local function could be a tiny bit confusing
- this function fprintf's an error, then its caller prints another one on failure

Reply:
Good point, I'll rename the function `get_flux_imp_path()`. 

I think both errors may be needed because if the path to the IMP is not set `flux_rpc_get_unpack()` will not return an error, but the `--with-imp` option should still fail.